docs/developer/architecture/kibana-platform-plugin-api.asciidoc

```diff
@@ -0,0 +1,350 @@
+[[kibana-platform-plugin-api]]
+== {kib} Plugin API
```

Comment:
@gchaps What are our recommendations regarding `{kib}` usage instead of Kibana? Should we always use `{kib}`? even in titles & tables?

Reply:
Yes, always use {kib}, even in titles and tables